rust/candid_parser/tests/assets/ok/typescript-interface/example.ts

```diff
@@ -0,0 +1,930 @@
+import { type HttpAgentOptions, type ActorConfig, type Agent, type ActorSubclass } from "@dfinity/agent";
+import type { Principal } from "@dfinity/principal";
+import { example as _example, createActor as _createActor, canisterId as _canisterId, CreateActorOptions } from "declarations/example";
+import { _SERVICE } from "declarations/example/example.did.d.js";
+export interface Some<T> {
+    __kind__: "Some";
+    value: T;
+}
+export interface None {
+    __kind__: "None";
+}
+export type Option<T> = Some<T> | None;
+function some<T>(value: T): Some<T> {
+    return {
+        __kind__: "Some",
+        value: value
+    };
+}
+function none(): None {
+    return {
+        __kind__: "None"
+    };
+}
+function isNone<T>(option: Option<T>): option is None {
+    return option.__kind__ === "None";
+}
+function isSome<T>(option: Option<T>): option is Some<T> {
+    return option.__kind__ === "Some";
+}
+function unwrap<T>(option: Option<T>): T {
+    if (isNone(option)) {
+        throw new Error("unwrap: none");
+    }
+    return option.value;
+}
+function candid_some<T>(value: T): [T] {
+    return [
+        value
+    ];
+}
+function candid_none<T>(): [] {
+    return [];
+}
+function record_opt_to_undefined<T>(arg: T | null): T | undefined {
+    return arg == null ? undefined : arg;
+}
+export type my_variant = {
+    /**
+     * Doc comment for my_variant field a
+    */
+    __kind__: "a";
+    a: {
```

Comment:
I would set the comment on the actual field instead of the "internal" kind field:

```suggestion
    __kind__: "a";
    /**
     * Doc comment for my_variant field a
    */
    a: {
```

Reply:
Make sense also to me.